packages/core/contracts/financial-templates/common/financial-product-libraries/KpiOptionsFinancialProductLibrary.sol

```diff
@@ -11,6 +11,42 @@ import "../../../common/implementation/Lockable.sol";
  * Post-expiry, the collateral requirement is left as 1 and the price is left unchanged.
  */
 contract KpiOptionsFinancialProductLibrary is FinancialProductLibrary, Lockable {
+    using FixedPoint for FixedPoint.Unsigned;
+
+    mapping(address => FixedPoint.Unsigned) financialProductTransformedPrices;
+
+    /**
+     * @notice Enables any address to set the transformed price for an associated financial product.
+     * @param financialProduct address of the financial product.
+     * @param transformedPrice the price for the financial product to be used if the contract is pre-expiration.
+     * @dev Note: a) Any address can set price transformations b) The price can't be set to blank.
+     * c) A transformed price can only be set once to prevent the deployer from changing it after the fact.
+     * d)  financialProduct must expose an expirationTimestamp method.
+     */
+    function setFinancialProductTransformedPrice(address financialProduct, FixedPoint.Unsigned memory transformedPrice)
```

Comment:
i dont really understand the full context of this change, but can you explain the consequences of allowing people to arbitrarily set this value?

Reply:
This would allow anyone to set the value for an EMP using this library, where before it was restricted to the deployer of the library (*not* the deployer of the EMP).

When this was restricted to the deployer of the library, that prevents anyone else from using the library without our help.

By opening it up to anyone, a deployer of a new EMP using this library can set a value after deployment but before putting any money in the EMP contract. It theoretically opens you up to an economically irrational griefing attack where someone sets a bogus value before you can set the real value, but we don't consider that a likely attack, and it could be mitigated by deploying the EMP and setting the transformed price in the same transaction if it actually started to happen.